src/i18n/arabic.rs

```diff
@@ -0,0 +1,144 @@
+extern crate alloc;
+use alloc::{string::String, vec};
+
+const NONE: u32 = u32::MAX;
+const AR_START: u32 = 0x621;
+const AR_END: u32 = 0x64A;
+const UNICODE_LAM: u32 = 0x644;
+
+pub fn use_contextual_forms(input: &str) -> String {
```

Comment:
@Konosh93 Digging around the 'net a bit, I found a library that covers this as well:
https://github.com/xclud/rust-bidi/blob/9c61091629072dc7c24b9a0f35c7a00a9906428a/src/letter_form.rs

However, your implementation is both more compact and elegant, so I'm inclined to keeping it. :)

Reply:
My rust repo search game is subpar, that would have saved me ours trying to recreate the C code in rust. 

Thanks for sharing 👍